rhc-ose-ansible/roles/secure-registry/tasks/main.yaml

```diff
@@ -0,0 +1,164 @@
+---
+
+  - name: "Check that Openshift Docker Registry exists"
+    command: "{{ openshift.common.client_binary }} get deploymentConfig {{ registry_dc }} -n {{ openshift_registry_project }}"
+    register: registry_exists
+    delegate_to: "{{ groups['masters'][0] }}"
+    run_once: true
+
+  - name: "fail"
+    fail: msg="No docker registry found in project default"
+    when: registry_exists.rc != 0
+    run_once: true
+    delegate_to: "{{ groups['masters'][0] }}"
+
+  - name: "fail"
+    fail: msg="Both registry_certificate and registry_key must be set, or neither"
+    when: (registry_certificate == "" and registry_key != "") or (registry_certificate != "" and registry_key == "")
+    delegate_to: "{{ groups['masters'][0] }}"
+    run_once: true
+
+  - name: "Collect registry service info"
+    command: "{{ openshift.common.client_binary}} get service {{ docker_registry_dc_name}} -n {{ openshift_registry_project }} -o yaml"
+    register: registry_svc_info
+    run_once: true
+    delegate_to: "{{ groups['masters'][0] }}"
+
+  - set_fact:
+      registry_svc_ip: "{{ (registry_svc_info.stdout | from_yaml).spec.clusterIP }}"
+    run_once: true
+    delegate_to: "{{ groups['masters'][0] }}"
+
+  - set_fact:
+      registry_svc_ports: ["{{item.port}}"]
+    with_items: "{{ (registry_svc_info.stdout | from_yaml).spec.ports}}"
+    run_once: true
+    delegate_to: "{{ groups['masters'][0] }}"
+
+  - name: "Creating certificate..."
+    command: "{{ openshift.common.admin_binary}} ca create-server-cert --signer-cert=/etc/origin/master/ca.crt \
```

Comment:
My idea was that it would re-secure the registry each time. However, yes i feel this would help idempotence. Which task should `creates:` be added to?